src/sinks/pulsar.rs

```diff
@@ -136,8 +156,20 @@ async fn healthcheck(producer: PulsarProducer) -> crate::Result<()> {
 
 impl PulsarSink {
     fn new(producer: PulsarProducer, encoding: EncodingConfig<Encoding>, acker: Acker) -> Self {
+        let schema = match &encoding.codec() {
+            Encoding::Avro => {
+                if let Some(schema) = &encoding.schema() {
+                    avro_rs::Schema::parse_str(schema).ok()
+                } else {
+                    None
```

Comment:
Hmmmm...  I think this `new` function should return a `crate::Result<Self>` now, this way we can return an error here if the user asks for avro without a schema.

```rust
    fn new(producer: PulsarProducer, encoding: EncodingConfig<Encoding>, acker: Acker) -> crate::Result<Self> {
        let schema = match &encoding.codec() {
            Encoding::Avro => {
                if let Some(schema) = &encoding.schema() {
                    avro_rs::Schema::parse_str(schema).ok()
                } else {
                    return Err("Avro requires a schema, specify a schema file with `encoding.schema`.".into())
                }
            }
            _ => None,
        };

        Ok(Self {
            encoding,
            avro_schema: schema,
            state: PulsarSinkState::Ready(Box::new(producer)),
            in_flight: FuturesUnordered::new(),
            acker,
            seq_head: 0,
            seq_tail: 0,
            pending_acks: HashSet::new(),
        })
    }
```

Reply:
Correct!